.github/workflows/modern-ci.yml

```
@@ -0,0 +1,202 @@
+          node_modules
+        key: deps-${{ runner.os }}-${{ hashFiles('composer.lock', 'package-lock.json') }}
+        restore-keys: deps-${{ runner.os }}-
+
+    - name: Install Dependencies
+      run: |
+        composer install --no-interaction --prefer-dist --optimize-autoloader
+        npm install
+        npm run build
+
+    - name: Setup Environment
+      run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2129:style:2:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects

(shellcheck)